clients/src/test/java/org/apache/kafka/clients/NetworkClientTest.java

```diff
@@ -81,6 +85,19 @@
     private final NetworkClient clientWithNoExponentialBackoff = createNetworkClient(reconnectBackoffMsTest);
     private final NetworkClient clientWithStaticNodes = createNetworkClientWithStaticNodes();
     private final NetworkClient clientWithNoVersionDiscovery = createNetworkClientWithNoVersionDiscovery();
+    private ArrayList<InetAddress> initialAddresses = new ArrayList<>(Arrays.asList(
```

Comment:
nit: Should we make these two static?

Reply:
Catching the `UnknownHostException` makes it a bit annoying, I threw it in a static block to make the compiler happy. Let me know if you think there's something neater.

Reply:
hi bob-barrett , I hava a quetion about your issue ,can you help me ? 
our kafka are deloying in k8s , where kafka cluster restart , result in producer and consumer will disconnect.
kafka cluster restart will generate new ips,  but kafka client store old ips . so we must restart producer and consumer , so tired.

in kafka client 2.8.1 version ,I see  org.apache.kafka.clients.NetworkClient#initiateConnect , `node.host()` maybe return a ip instead of domain.   if return a ip , will result in lose efficacy about your updated version .

my english level is match awful ， can you know my words?  

```
private void initiateConnect(Node node, long now) {
        String nodeConnectionId = node.idString();
        try {
            connectionStates.connecting(nodeConnectionId, now, node.host(), clientDnsLookup);
            InetAddress address = connectionStates.currentAddress(nodeConnectionId);
            log.debug("Initiating connection to node {} using address {}", node, address);
            selector.connect(nodeConnectionId,
                    new InetSocketAddress(address, node.port()),
                    this.socketSendBuffer,
                    this.socketReceiveBuffer);
        } catch (IOException e) {
            log.warn("Error connecting to node {}", node, e);
            // Attempt failed, we'll try again after the backoff
            connectionStates.disconnected(nodeConnectionId, now);
            // Notify metadata updater of the connection failure
            metadataUpdater.handleServerDisconnect(now, nodeConnectionId, Optional.empty());
        }
    }
```